### DISCUSSION TITLE
Cannot access 'Model' before initialization

### DISCUSSION THREAD

**Discussion by truongbo17** [adonisjs/core#4620] [Help]
my controller:
import {inject} from "@adonisjs/fold";

@inject()
export default class TradeController {
  public constructor(private symbolService: SymbolService) {
  }

public async getListSymbols({response}: HttpContextContract) {
console.log(await this.symbolService.getSymbol())
}
}

my service:
@inject()
export default class SymbolService {
  constructor(private symbolRepository: SymbolRepository) {
  }
public async getSymbol() {
return this.symbolRepository.getActiveSymbols()
}
}

my repo:
import Symbol from "App/Models/Symbol";
import {SymbolStatus} from "App/Enum/SymbolStatus";

export default class SymbolRepository implements SymbolRepositoryInterface {
  public async getActiveSymbols(): Promise<any> {
    const symbol = await Symbol.query().where('status', SymbolStatus.ACTIVE)
    return Promise.resolve(symbol);
  }
}

and error::[11:00:26.844] ERROR (chatapp/4161923): Cannot access 'Symbol' before initialization

![image](https://github.com/adonisjs/core/assets/91400591/aaef6fc4-f2bf-42ca-a17e-ca1bfda47a4b)

**Comment by RomainLanz** [comment]
Hey @truongbo17! 👋🏻 

Are you importing `SymbolRepository` anywhere else in your codebase? For example, in a provider.
It looks like the class was imported before the framework had time to boot.
